src/rrdp.rs

```diff
@@ -1774,4 +1784,31 @@ mod test {
             )
         );
     }
+
+    #[test]
+    fn has_matching_origins() {
+        fn check<const N: usize>(
+            snapshot: &str, deltas: [&str; N]
+        ) -> bool {
+            let hash = Hash::from_data(b"12");
+            NotificationFile::new(
+                Uuid::nil(), 0,
+                SnapshotInfo::new(
+                    uri::Https::from_str(snapshot).unwrap(), hash
+                ),
+                deltas.iter().map(|uri| {
+                    DeltaInfo::new(
+                        0, uri::Https::from_str(uri).unwrap(), hash
+                    )
+                }).collect()
+            ).has_matching_origins(
+                &uri::Https::from_str("https://foo.bar/n/o").unwrap()
+            )
+        }
+
+        assert!(check(
+            "https://foo.bar/1/2/3",
+            ["https://foo.bar/", "https://foo.bar/4", "https://foo.bar/7/8"],
+        ));
+    }
```

Comment:
Even though trivial, perhaps test the failing case as well? A la

```suggestion
        ));
        
        assert!(!check(
            "https://foo.bar/1/2/3",
            ["https://foo.bar/", "https://foo.local/4"],
        ));    
    }
```